packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/palette/DiagramPaletteContext.types.ts

```diff
@@ -17,6 +17,8 @@ export interface DiagramPaletteContextValue {
   isOpened: boolean;
   showDiagramPalette: (x: number, y: number) => void;
   hideDiagramPalette: () => void;
+  getLastToolInvoked: (toolSectionId: string) => string | undefined;
+  setLastToolInvoked: (toolSectionId: string, toolId: string) => void;
 }
```

Comment:
Wouldn't it be more accurate to call these methods `getLastToolIdInvoked` and `setLastToolIdInvoked` ?

Reply:
Actually I'll see if I can put the actuall `GQLTool` here instead of just its id.

Reply:
Done.